webtau-http/src/main/java/org/testingisdocumenting/webtau/http/datanode/DataNodeCompareToHandler.java

```diff
@@ -89,6 +90,6 @@ private Object extractActual(DataNode actual) {
             return actual.elements();
         }
 
-        return actual.asMap();
+        return DataNodeToMapConverter.convertToMap(actual);
```

Comment:
Ideally this would also convert to something that would track `has` and `get` calls on the node itself.  Does the caller of this function really need a map or could they work with the data node directly?

Reply:
updated code and added comment and a test